### PR TITLE
Visual effects for being close to death

### PR DESCRIPTION
**Pull request by Sajdkick**
Added a script called StressScript, the purpose of this script is to induce stress into the player when the monster gets closer. At the moment you can control the volume and light intensity as the monster gets closer.

### CODE REVIEW

File: src/Engine/Manager/ScriptManager.cpp [hunk 1]
```diff
@@ -1000,13 +1000,13 @@ void ScriptManager::LoadScriptFile(const char* fileName, std::string& script){
     fclose(f);
 }
 
-void ScriptManager::ExecuteCall(asIScriptContext* context) {
+void ScriptManager::ExecuteCall(asIScriptContext* context, std::string scriptName) {
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MINOR Function parameter 'scriptName' should be passed by reference. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3ApassedByValue)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dd6f97d89504a78257ff2d2e009f9f2aa5b4cc20/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dd6f97d89504a78257ff2d2e009f9f2aa5b4cc20/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L609](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dd6f97d89504a78257ff2d2e009f9f2aa5b4cc20/src/Engine/Manager/ScriptManager.cpp#L609): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L851](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dd6f97d89504a78257ff2d2e009f9f2aa5b4cc20/src/Engine/Manager/ScriptManager.cpp#L851): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L857](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dd6f97d89504a78257ff2d2e009f9f2aa5b4cc20/src/Engine/Manager/ScriptManager.cpp#L857): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/ScriptManager.cpp [hunk 2]
```diff
@@ -1000,13 +1000,13 @@ void ScriptManager::LoadScriptFile(const char* fileName, std::string& script){
     fclose(f);
 }
 
-void ScriptManager::ExecuteCall(asIScriptContext* context) {
+void ScriptManager::ExecuteCall(asIScriptContext* context, std::string& scriptName) {
```
Comment (Chainsawkitten) [inline src/Engine/Manager/ScriptManager.cpp]: Should be a const reference

File: Game/Resources/StressScript.as [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class StressScript {
```
Comment (Chainsawkitten) [inline Game/Resources/StressScript.as]: Inconsistent indentation (tabs vs. spaces).

File: Game/Resources/MonsterTestScript.as [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class MonsterTestScript {
```
Comment (Chainsawkitten) [inline Game/Resources/MonsterTestScript.as]: Inconsistent indentation (tabs vs. spaces).

Comment (Sajdkick) [reply Game/Resources/MonsterTestScript.as]: Fixed

File: Game/Resources/StressScript.as (hunk 3 above)
Comment (Sajdkick) [reply Game/Resources/StressScript.as]: Fixed

**Pull request merged**